app/routes/github.js

```diff
@@ -28,30 +28,30 @@ export default Route.extend(AuthenticatedRouteMixin, {
     let stateValidator = get(this, 'githubState');
 
     if (stateValidator.validate(state)) {
-      return this._sendRequest(code);
+      return this._sendRequest(code, state);
```

Comment:
The idea is

- We hit the github authorize URL with a `redirect_to` param and a `state` param
- On the authorization URL, if the user accepts, we are redirected back to whichever page is specified in `redirect_to`,  with a `code` param and a `state` param. 
- Our app is then obligated to check the `state` param against the original. Since this is a cross-site navigation, we need to store the original somehow and compare against it. In our case, this is done using the session data facilities provided by `ember-simple-auth` (which internally uses local storage), abstracted through our `github-state` service. In other cases, it could be some other form of local storage, a cookie, or a server-stored value.